### PR TITLE
Traefik mtls

### LINKED ISSUE #9828: [stable/traefik] Support mutual TLS authentication
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): FEATURE REQUEST

**Version of Helm and Kubernetes**:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-04T07:51:55Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

$ helm version
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```

**Which chart**: stable/traefik

**What happened**: Unable to configure traefik to enable [mutual TLS authentication](https://docs.traefik.io/configuration/entrypoints/#tls-mutual-authentication)

**What you expected to happen**: The chart should provide the ability to enable and configure mTLS

**How to reproduce it** (as minimally and precisely as possible):

1. Deploy traefik
2. Deploy a HTTPS web server that echoes received headers back to the client, such as [echoheaders](https://github.com/kubernetes/contrib/tree/master/ingress/echoheaders)
3. Access the web server, providing a client certificate: `curl --cert tls.crt --key tls.key https://myserver.example.com`
4. Notice there is no "x-forwarded-tls-client-cert" header in the returned response body
5. Notice that it is possible to access the web server without using mTLS: `curl https://myserver.example.com`

**Anything else we need to know**:
No

### PR DESCRIPTION
**Pull request by tlm**
#### What this PR does / why we need it:
Adds support to the Traefik helm chart to support its already existing MTLS settings through helm values. This support is needed so Traefik can be deployed into Kubernetes using helm and support mutual TLS auth.

#### Which issue this PR fixes
#9828 

#### Special notes for your reviewer:
The original work was done by @somcsel

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/values.yaml [hunk 1]
```diff
@@ -14,8 +14,7 @@ whiteListSourceRange: []
 externalTrafficPolicy: Cluster
 replicas: 1
 # startupArguments:
-#   - "--ping"
-#   - "--ping.entrypoint=http"
+#   - "--pitrusk   - "--ping.entrypoint=http"
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: pitrusk eh?

File: stable/traefik/values.yaml [hunk 2]
```diff
@@ -114,6 +113,14 @@ ssl:
     # basic:
     #   testuser: $apr1$JXRA7j2s$LpVns9vsme8FHN0r.aSt11
 
+  # mtls:
+  #   enabled: true
+  #   optional: false
+  #   clientCaCerts: ["wef"]
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: Not sure what wef is...

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/templates/_helpers.tpl [hunk 3]
```diff
@@ -123,3 +123,14 @@ Create the block for RootCAs.
 	   {{- end -}}
          ]
 {{- end -}}
+
+Create the block for mTLS ClientCAs.
```
Comment (dtomcej) [inline stable/traefik/templates/_helpers.tpl:129]: I think you are missing an open comment block here

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/templates/_helpers.tpl (hunk 3 above)
Comment (tlm) [reply stable/traefik/templates/_helpers.tpl:129]: Good spot, looks like the block above this one was also missing the same. Have added both

**Comment by dtomcej** [comment]
Awesome job. This PR looks good to go.

Just needs a Chart version bump, and we are 🚢!

**Comment by tlmiller** [comment]
@dtomcej  Cheers Done

**Comment by mmisztal1980** [comment]
is this PR getting completed any time soon?

**Pull request merged**